### PR TITLE
Update Freebayes- Trying newer simple build

### PR DESCRIPTION
**Pull request by jpuritz**
Freebayes now used the meson and ninja-build system and can rely on installed packages.  This should simplify the build and no longer rely on a recursive git download.
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by jpuritz** [comment]
@jts I put this up as a start for you.  I think this should get close to working.

**Comment by jpuritz** [comment]
I'm not familiar with meson or ninja-build, so this may be beyond my capabilities.  However, if we this makes things simpler major BIOCONDA overhauls that would be great.

**Comment by jts** [comment]
Thanks everyone, I'm in meetings for the rest of the day so won't be able to help today but will keep an eye on this and pitch in when I can.

File: recipes/freebayes/meta.yaml [hunk 1]
```diff
@@ -20,18 +21,23 @@ requirements:
...
   run:
     - zlib
     - bzip2
     - python
     - htslib
     - parallel
     - xz
+    - bc
+    - samtools
+    - meson
```
Comment (bgruening) [inline recipes/freebayes/meta.yaml:34]: meson and nija should probably be in the host section, isn't it?

Comment (jpuritz) [reply recipes/freebayes/meta.yaml:34]: https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pbmm2/meta.yaml has in in the build section, but yes I think you are right.  I'm a few months rusty doing this.

**Comment by jpuritz** [comment]
@pjotrp I've set this up, and I think should be the basis for a new working build via bioconda.  Please feel free to contribute directly.

**Comment by pjotrp** [comment]
The CI takes 5 hours to start compiling(!?). Not sure why it stops loading a Docker image.

**Comment by jpuritz** [comment]
Well, it no longer takes 5 hours, but `meson` seems to not think that `$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc` is a compiler:

```
14:58:02 BIOCONDA INFO (OUT) meson.build:9:0: ERROR: Compiler $BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc can not compile programs.
```

**Comment by jpuritz** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
Nicely done @jpuritz !

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Comment by jts** [comment]
Thanks for all your work on this @jpuritz, much appreciated.

**Comment by jpuritz** [comment]
Thanks @dpryan79 and @jts.  This build has always drove me crazy.  So happy that it can be this simple.  @dpryan79 I just saw that I left some commented out old code in the `build.sh` file.  I will clean that up.  

@pjotrp and @jts I think this format can now be followed for vcflib as well, right?

**Comment by pjotrp** [comment]
Thanks a ton!! @ekg will be very happy. I'll make sure the badges are up-to-date soon.

@jpuritz vcflib is still using cmake and should work. The Debian team just included that one too.

**Comment by dancooke** [comment]
@pjotrp @jpuritz It looks like since this PR, `freebayes-parallel` and other scripts no longer makes their way to `$PREFIX/bin` so are no longer accessible from `PATH` immediately after `conda install freebayes`. Is this intended?

**Comment by jpuritz** [comment]
#27011 should now fix this if all builds and tests well.

**Pull request merged**